File: 1_top.c

#include <stdio.h>
#include <stdlib.h>

int* twoSum(int* nums, int numsSize, int target) {
    int _[100001] = {0}, *index_plus_one = _ + 50000;

    for (int i = 0; i < numsSize; i++) {
        int rest = target - nums[i];

        if (index_plus_one[rest]) {
            int *ans = malloc(sizeof(int) * 2);
            ans[0] = i;
            ans[1] = index_plus_one[rest] - 1;
            return ans;
        } else
            index_plus_one[nums[i]] = i + 1;
    }

    return NULL;
}

int main(int argc, char **argv) {
    int nums[4]= {2, 7, 11, 15};
    int target=13;

    int *res=twoSum(nums, 4, target);

    if(res) {
        for(int i=0; i<sizeof(res)/sizeof(res[0]); i++) {
            printf("%d", *(res+i));
        }
    }

    return 0;

}

File: 1_top.py
import os
import sys

class Solution(object):
    def twoSum(self, nums, target):
        if len(nums)<=1:
            return False
        buff_dict={}

        for i in range(len(nums)):
            if nums[i] in buff_dict:
                return [buff_dict[nums[i]], i+1]
            else:
                buff_dict[target-nums[i]]=i+1

def main(argv):
    nums=[1, 3, 6, 8, 14, 19, 20, 23, 25]
    target=23

    solution=Solution()
    res=solution.twoSum(nums, target)

    for i in res:
        print i

if __name__=='__main__':
    reload(sys)
    sys.setdefaultencoding('utf-8')
    main(sys.argv)

